hs_core/models.py

```diff
@@ -1109,7 +1059,128 @@ def update(cls, element_id, **kwargs):
+        else:
+            graph.add((relation_node, getattr(DCTERMS, self.type), Literal(self.value)))
+
+    @classmethod
+    def ingest_rdf(cls, graph, subject, content_object):
+        for _, _, relation_node in graph.triples((subject, cls.get_class_term(), None)):
+            for _, p, o in graph.triples((relation_node, None, None)):
+                type_term = p
+                value = o
+                break
+            if type_term:
+                type = type_term.split('/')[-1]
+                value = str(value)
+                Relation.create(type=type, value=value, content_object=content_object)
+
+
+@rdf_terms(HSTERMS.geospatialRelation, text=HSTERMS.relation_name)
+class GeospatialRelation(AbstractRelation):
+
+    SOURCE_TYPES = (
+        (RelationTypes.relation.value, 'The content of this resource is related to'),
+    )
+
+    term = 'GeospatialRelation'
+    type = models.CharField(max_length=100, choices=SOURCE_TYPES)
+    value = models.TextField()
+    text = models.TextField(max_length=100)
+
+    def rdf_triples(self, subject, graph):
+        relation_node = BNode()
+        graph.add((subject, self.get_class_term(), relation_node))
+        graph.add((relation_node, getattr(DCTERMS, self.type), URIRef(self.value)))
+        graph.add((relation_node, HSTERMS.relation_name, Literal(self.text)))
+
+    def check_text(relation):
+        relative_id = relation.value.split("ref/").pop()
+        collection = relative_id.split("/")[0]
+        id = relative_id.split("/")[1]
+        url = f"https://reference.geoconnex.us/collections/{collection}/items/{id}?" \
+               "f=jsonld&lang=en-US&skipGeometry=true"
+
+        response = urllib.request.urlopen(url)
+        str_response = response.read()
+        str_response = json.loads(str_response)
+        contexts = str_response['@context']
+        for context in contexts:
+            compacted = jsonld.compact(str_response, context)
+            try:
+                name = compacted['schema:name']
+            except KeyError:
+                continue
+            text = f"{name} [{relative_id}]"
+            if relation.text != text:
+                print(f"Updating {relation.value}, '{relation.text}' to '{text}'.")
+                relation.text = text
+                relation.save()
+            else:
+                print(f"Not updating relation '{relation.value}'. Geoconnex API matches HS.")
+
+    @classmethod
+    def sync_all_text(cls):
+        relations = cls.objects.all()
+        for relation in relations:
+            relation.check_text()
+
+    @classmethod
+    def ingest_rdf(cls, graph, subject, content_object):
+        for _, _, relation_node in graph.triples((subject, cls.get_class_term(), None)):
+            type = value = name = None
+            for _, p, o in graph.triples((relation_node, None, None)):
+                if p == HSTERMS.relation_name:
```

Comment:
This field term is declared in the rdf_terms decorator.  Pull the term from there in case it changes.

```
text_field_term = cls.get_field_term("text")
```

Reply:
updated, thanks